### PR TITLE
fixed Deleting state

### PR DESCRIPTION
**Pull request by oneilcin**
DeleteCluster was slow in AWS which revealed it was displaying the Created/Ready state while delete was happening.  I changed the logic to Display the Deleting State before the blocking call to ClusterDestroy.  Only a cluster that has a healthy juju controller (JujuBootstrapReady) can be deleted with ClusterDestroy().

### CODE REVIEW

File: assets/clusterrole.yaml [hunk 1]
```diff
@@ -7,3 +7,16 @@ rules:
...
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  # "namespace" omitted since ClusterRoles are not namespaced
+  name: krakencluster-writer
+rules:
+- apiGroups: ["samsung.cnct.com"]
+  resources: ["krakenclusters", "events", "namespaces"]
+  verbs: ["create", "update", "delete","edit","patch"]
```
Comment (alejandroEsc) [inline assets/clusterrole.yaml]: ['*'] or quotes should be fine for verbs.

File: assets/clusterrolebinding.yaml [hunk 2]
```diff
@@ -10,3 +10,16 @@ roleRef:
...
   name: krakencluster-reader
   apiGroup: rbac.authorization.k8s.io 
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: krakencluster-writer 
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (alejandroEsc) [inline assets/clusterrolebinding.yaml]: namespace etc.. should be configurable and depend on helm.

**Comment by leahnp** [comment]
@alejandroEsc @oneilcin is it easier to merge mine and rebase this one? Or vis-versa. Let me know what's easiest and how I can help. Otherwise - LGTM.

**Comment by oneilcin** [comment]
I just rebased this.  When CI is ready, it is good to Merge

**Pull request merged**